### PR TITLE
Explain how to do SQL CONCAT() better...

### PR DESCRIPTION
**Pull request by ionas**
... as there is no smooth way to do virtual fields across multiple entities.

### CODE REVIEW

**Comment by ionas** [comment]
Taken from my IRC helping gist: https://gist.github.com/anonymous/669478df7a59410b6240bdf45f47a59a

**Comment by davidyell** [comment]
I've never got this, when `select(['name' => 'CONCAT(first_name, ' ', last_name)'])` is so much easier.

**Comment by svaldenegro** [comment]
I think `select(['name' => 'CONCAT(first_name, ' ', last_name)'])` is not optimal if you want to port the application.

**Comment by ionas** [comment]
@svaldenegro across different SQL adapters. Yep.

File: en/orm/query-builder.rst [hunk 1]
```diff
@@ -336,12 +336,14 @@ you can use, literal arguments and bound parameters. Literal parameters allow
 you to reference columns or other SQL literals. Bound parameters can be used to
 safely add user data to SQL functions. For example::
 
-    $query = $articles->find();
+    $query = $articles->find()->innerJoinWith('Categories');
     $concat = $query->func()->concat([
-        'title' => 'literal',
-        ' NEW'
+        'Articles.title' => 'literal',
```
Comment (lorenzo) [inline en/orm/query-builder.rst]: Can you change `literal` with `identifier` please?

Comment (ionas) [inline en/orm/query-builder.rst]: Sure I can, I was not aware that this got changed/added.

**Comment by ionas** [comment]
Please double check.

**Comment by ionas** [comment]
@lorenzo since when is `identifier` available instead of just `literal` (I can add a versionadd comment thus that old version users don't get screwed)

**Comment by davidyell** [comment]
If you can use a virtual property, then the example needs changing, so that people don't see it and think that it's the way to go when an entity mutator would be a better option.

However, the counter-point is that you can't use an entity mutator in a query, so I guess it's valid in that use-case.

**Pull request merged**